Formula/cython.rb

```diff
@@ -19,21 +20,32 @@ class Cython < Formula
     Users are advised to use `pip` to install cython
   EOS
 
-  depends_on "python@3.9"
+  depends_on "python@3.8" => [:build, :test]
+  depends_on "python@3.9" => [:build, :test]
+  depends_on "python@3.10"
```

Comment:
The idea of this PR is fine with me, though I don't think there is a way around the audit failure while `python@3.10` is a runtime dep.

Reply:
Yeah initially I thought I could make it a build/test dep, bit it is still needed for the cython binary.

I can add a new allowlist in brew I guess?

Reply:
> I can add a new allowlist in brew I guess?

Yes, please!

Reply:
Another alternative to this, in the case of `cython`, since it ships executables, is to provide separate Cython formulae which each depend on a different version of Python.

I'm of two minds about this: this would lead to multiple formulae which build from the same source. On the other hand, it might provide a better user experience (e.g. we could make at least one of these formulae not-keg-only since the keg-only status of `cython` is really from the time we didn't want to package anything you could `pip install`).

Reply:
To be clear on this, though: I am definitely against multiple `numpy` formulae, since it does not ship executables, and the builds that result from different Python versions do not conflict.